boards/arm/pinetime_devkit0/key_out.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2021 Casper Meijn <casper@meijn.net>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/gpio.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(pine64_pinetime_key_out);
+
+#define KEY_OUT_NODE DT_INST(0, pine64_pinetime_key_out)
+
+#if DT_NODE_HAS_STATUS(KEY_OUT_NODE, okay)
+
+static const struct gpio_dt_spec key_out = GPIO_DT_SPEC_GET_OR(KEY_OUT_NODE, gpios, { 0 });
```

Comment:
Hard-coding DT_INST numbers like this is generally a bad practice that I am trying to avoid doing more in mainline zephyr, and in this case it looks a bit suspicious.

I say "mostly" because this is a file in the board directory, so you know for sure that the key_out node exists. So my questions are:

- Why not just use DT_PATH(key_out) to get a node identifier, since you know the node exists, and where it is?
- Why not use GPIO_DT_SPEC_GET? That won't fail, because you know the node has the right property

You can hard-code both the node and its property here in board-specific code, is basically what I am saying.

Reply:
>     * Why not just use DT_PATH(key_out) to get a node identifier, since you know the node exists, and where it is?

Because it didn't know this is bad practice. To me `DT_INST` feels flexible, as you are not dependent on the name of the node. I changed it to the suggested DT_PATH.

>     * Why not use GPIO_DT_SPEC_GET? That won't fail, because you know the node has the right property

You are right. If the node exists, then I know it has the right property. I removed the `_OR` and the if-statement.

Reply:
> `DT_INST` feels flexible, as you are not dependent on the name of the node

Right, but you also have no idea which node you are getting if there is more than one. Often, people don't realize that instance numbers have few restrictions on their order. That's why I generally think hard-coded instance numbers are a bad idea in general, as they often lead to mistakes that just happen to work for the one developer's build when they were writing the code that day. There are helpers like `DT_COMPAT_GET_ANY_STATUS_OKAY` (as well as higher level related helpers like `DEVICE_DT_GET_ANY` and `DEVICE_DT_GET_ONE`) for situations where you really don't care which one you are getting, like in samples.

Hopefully that makes sense, thanks for making the change.